File: c/Estruturas Condicionais/troco.c
#include <stdio.h>

int main() {
    double valorCompra, valorRecebido, troco;
    int quantidadeProdutos;

    printf("Digite o valor do produto: ");
    scanf("%lf", &valorCompra);
    printf("Quantos produtos foram comprados? ");
    scanf("%d", &quantidadeProdutos);

    valorCompra *= quantidadeProdutos;
    printf("Qual o valor pago? ");
    scanf("%lf", &valorRecebido);

    troco = valorRecebido - valorCompra;

    if (troco < 0) {
        printf("Valor recebido e insuficiente.\n");
    } else if (troco == 0) {
        printf("O valor pago e exato. Nao ha troco.\n");
    } else {
        printf("O troco e: R$ %.2lf\n", troco);
    }

    return 0;
}

File: java/Estruturas condicionais/Troco.java
import java.util.Locale;
import java.util.Scanner;

public class Troco {

    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int qnt;
        double preco, recebido, troco;

        System.out.print("Qual o preço do produto? ");
        preco = sc.nextDouble();

        System.out.print("Qual a quantidade do produto? ");
        qnt = sc.nextInt();

        System.out.print("Qual o valor recebido? ");
        recebido = sc.nextDouble();

        troco = recebido - (preco * qnt);

        if (troco >= 0) {
            System.out.println("TROCO = " + troco + " Reais");
        } else {
            System.out.println("DINHEIRO INSUFICIENTE. FALTAM " + Math.abs(troco) + " Reais");
        }

        sc.close();
    }
}
